submissions/madmaxWMFU/Friends-App/js/app.js

```diff
@@ -0,0 +1,100 @@
+                    </div> 
+                    <span class="user-card__age">${user.dob.age}</span>
+                </div>`;
+    friendZone.insertAdjacentHTML("beforeEnd", temp);
+}
+
+const cleanFriendsZone = () => friendZone.innerHTML = "";
+
+const renderFriendsList = (allFriends) => allFriends.forEach(friend => drawFriendsCards(friend));
+
+const funcFilterList = {
+    filterByGender(list, {gender}) { return gender !== "all" ? list.filter((person) => person.gender === gender) : allFriends.currentList},
+    filterByAge(list, {ageTill, ageTo}) { return list.filter((person) => ageTill <= person.dob.age && person.dob.age <= ageTo)},
+    filterByName(list, {name}) { return name != "" ? list.filter((person) => new RegExp(name, 'i').test(person.name.first)) : null}
+}
+
+const getFilterList = () => {
+    allFriends.changeList = [...allFriends.currentList];
+    const searchValue = {
+        name: document.querySelector(".search-input-name").value || "",
+        gender: document.querySelector("[name='search-input-gender']:checked").value || "all",
+        ageTill: document.querySelector(".search-input-age1").value || allFriends.minAge,
+        ageTo: document.querySelector(".search-input-age2").value || allFriends.maxAge
+    };
+    let funcName = ["filterByName", "filterByGender", "filterByAge"];
+
+    funcName.forEach((func) => {
+        let newArray = funcFilterList[func](allFriends.changeList, searchValue);
+        if(newArray != null) allFriends.changeList = newArray;
+    });
+
+    cleanFriendsZone();
+    renderFriendsList(allFriends.changeList);  
+}
+
+const getSortList = ({target}) => {
+    switch(target.value) {
+        case "nameAsc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => a.name.first < b.name.first ? -1 : 1);
+            break;
+        case "nameDesc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => a.name.first > b.name.first ? -1 : 1);
+            break;
+        case "ageAsc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => a.dob.age - b.dob.age);
+            break;
+        case "ageDesc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => b.dob.age - a.dob.age);
+            break;
+    }
+    cleanFriendsZone();
+    renderFriendsList(allFriends.changeList);
+};
+
+const resetSearchValues = () => {
+    filters.options[0].selected = true;
+    document.querySelector(".search-input-name").value = "";
+    document.querySelector(".search-input-age1").value = allFriends.minAge;
+    document.querySelector(".search-input-age2").value = allFriends.maxAge;
+    document.querySelectorAll("[name='search-input-gender']")[2].checked = true;
+    renderFriendsList(allFriends.currentList);
+};
+
+filters.addEventListener("click", getSortList);
+sorts.addEventListener("change", getFilterList);
+resetFilters.addEventListener("click", resetSearchValues);
+
+window.onload = function() {
+    getDataApi()
```

Comment:
It is a good place to make realization of pure functions with single responsibilty, etc.